Android: Error when starting navigation in landscape

I have my orientation for my MapView Activity set to `Forced Portrait`, when i turn the device into `Landscape` (_my activity is still in potrait - this is what i expect_).

When I start the NavigationUI in this way I get the below error:

**Android Version:** 4.1.2
**Device:*** Motorola TC55 

```
10-09 14:06:58.334 15284-15284/uk.co.ainscough.routeinfo E/AndroidRuntime: FATAL EXCEPTION: main
                                                                           java.lang.RuntimeException: Unable to start activity ComponentInfo{uk.co.ainscough.routeinfo/com.mapbox.services.android.navigation.ui.v5.NavigationView}: java.lang.NullPointerException
                                                                               at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2108)
                                                                               at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2133)
                                                                               at android.app.ActivityThread.access$600(ActivityThread.java:142)
                                                                               at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1234)
                                                                               at android.os.Handler.dispatchMessage(Handler.java:99)
                                                                               at android.os.Looper.loop(Looper.java:137)
                                                                               at android.app.ActivityThread.main(ActivityThread.java:4895)
                                                                               at java.lang.reflect.Method.invokeNative(Native Method)
                                                                               at java.lang.reflect.Method.invoke(Method.java:511)
                                                                               at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:789)
                                                                               at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:556)
                                                                               at dalvik.system.NativeStart.main(Native Method)
                                                                            Caused by: java.lang.NullPointerException
                                                                               at android.support.v4.graphics.drawable.DrawableWrapperApi14.getState(DrawableWrapperApi14.java:155)
                                                                               at android.support.v4.graphics.drawable.DrawableWrapperApi14.setTintList(DrawableWrapperApi14.java:278)
                                                                               at android.support.v4.graphics.drawable.DrawableWrapperApi14.setTint(DrawableWrapperApi14.java:272)
                                                                               at android.support.v4.graphics.drawable.DrawableCompat$DrawableCompatBaseImpl.setTint(DrawableCompat.java:68)
                                                                               at android.support.v4.graphics.drawable.DrawableCompat.setTint(DrawableCompat.java:421)
                                                                               at com.mapbox.services.android.navigation.ui.v5.instruction.InstructionView.initBackground(InstructionView.java:249)
                                                                               at com.mapbox.services.android.navigation.ui.v5.instruction.InstructionView.onFinishInflate(InstructionView.java:106)
                                                                               at android.view.LayoutInflater.rInflate(LayoutInflater.java:754)
                                                                               at android.view.LayoutInflater.rInflate(LayoutInflater.java:749)
                                                                               at android.view.LayoutInflater.inflate(LayoutInflater.java:489)
                                                                               at android.view.LayoutInflater.inflate(LayoutInflater.java:396)
                                                                               at android.view.LayoutInflater.inflate(LayoutInflater.java:352)
                                                                               at android.support.v7.app.AppCompatDelegateImplV9.setContentView(AppCompatDelegateImplV9.java:287)
                                                                               at android.support.v7.app.AppCompatActivity.setContentView(AppCompatActivity.java:139)
                                                                               at com.mapbox.services.android.navigation.ui.v5.NavigationView.onCreate(NavigationView.java:87)
                                                                               at android.app.Activity.performCreate(Activity.java:5042)
                                                                               at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1082)
                                                                               at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2072)
                                                                               at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2133) 
                                                                               at android.app.ActivityThread.access$600(ActivityThread.java:142) 
                                                                               at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1234) 
                                                                               at android.os.Handler.dispatchMessage(Handler.java:99) 
                                                                               at android.os.Looper.loop(Looper.java:137) 
                                                                               at android.app.ActivityThread.main(ActivityThread.java:4895) 
                                                                               at java.lang.reflect.Method.invokeNative(Native Method) 
                                                                               at java.lang.reflect.Method.invoke(Method.java:511) 
                                                                               at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:789) 
                                                                               at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:556) 
                                                                               at dalvik.system.NativeStart.main(Native Method) 
10-09 14:06:58.344 631-631/? W/ActivityManager:   Force finishing activity uk.co.ainscough.routeinfo/com.mapbox.services.android.navigation.ui.v5.NavigationView
```

--- a/libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/instruction/InstructionView.java
+++ b/libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/instruction/InstructionView.java
@@ -2,6 +2,7 @@ package com.mapbox.services.android.navigation.ui.v5.instruction;
 
 import android.arch.lifecycle.Observer;
 import android.content.Context;
+import android.content.res.Configuration;
 import android.graphics.drawable.Drawable;
 import android.os.Bundle;
 import android.os.Parcelable;
@@ -56,7 +57,6 @@ import java.text.DecimalFormat;
  */
 public class InstructionView extends RelativeLayout {
 
-  private View instructionLayout;
   private ImageView maneuverImage;
   private TextView stepDistanceText;
   private TextView stepInstructionText;
@@ -233,7 +233,6 @@ public class InstructionView extends RelativeLayout {
     stepInstructionText = findViewById(R.id.stepInstructionText);
     soundChipText = findViewById(R.id.soundText);
     soundFab = findViewById(R.id.soundFab);
-    instructionLayout = findViewById(R.id.instructionLayout);
     rerouteLayout = findViewById(R.id.rerouteLayout);
     turnLaneLayout = findViewById(R.id.turnLaneLayout);
     rvTurnLanes = findViewById(R.id.rvTurnLanes);
@@ -245,8 +244,18 @@ public class InstructionView extends RelativeLayout {
       int navigationViewPrimaryColor = ThemeSwitcher.retrieveNavigationViewPrimaryColor(getContext());
       int navigationViewSecondaryColor = ThemeSwitcher.retrieveNavigationViewSecondaryColor(getContext());
       // Instruction Layout - primary
-      Drawable instructionBackground = DrawableCompat.wrap(instructionLayout.getBackground()).mutate();
-      DrawableCompat.setTint(instructionBackground, navigationViewPrimaryColor);
+      if (getContext().getResources().getConfiguration().orientation == Configuration.ORIENTATION_LANDSCAPE) {
+        View instructionLayoutText = findViewById(R.id.instructionLayoutText);
+        View instructionLayoutManeuver = findViewById(R.id.instructionLayoutManeuver);
+        Drawable textBackground = DrawableCompat.wrap(instructionLayoutText.getBackground()).mutate();
+        Drawable maneuverBackground = DrawableCompat.wrap(instructionLayoutManeuver.getBackground()).mutate();
+        DrawableCompat.setTint(textBackground, navigationViewPrimaryColor);
+        DrawableCompat.setTint(maneuverBackground, navigationViewPrimaryColor);
+      } else {
+        View instructionLayout = findViewById(R.id.instructionLayout);
+        Drawable instructionBackground = DrawableCompat.wrap(instructionLayout.getBackground()).mutate();
+        DrawableCompat.setTint(instructionBackground, navigationViewPrimaryColor);
+      }
       // Sound chip text - primary
       Drawable soundChipBackground = DrawableCompat.wrap(soundChipText.getBackground()).mutate();
       DrawableCompat.setTint(soundChipBackground, navigationViewPrimaryColor);

--- a/libandroid-navigation-ui/src/main/res/layout-land/instruction_layout.xml
+++ b/libandroid-navigation-ui/src/main/res/layout-land/instruction_layout.xml
@@ -12,7 +12,7 @@
     tools:showIn="@layout/instruction_view_layout">
 
     <LinearLayout
-        android:id="@+id/linearLayout"
+        android:id="@+id/instructionLayoutManeuver"
         android:layout_width="wrap_content"
         android:layout_height="wrap_content"
         android:layout_marginBottom="4dp"
@@ -44,14 +44,14 @@
     </LinearLayout>
 
     <FrameLayout
-        android:id="@+id/frameLayout"
+        android:id="@+id/instructionLayoutText"
         android:layout_width="0dp"
         android:layout_height="wrap_content"
         android:layout_marginTop="0dp"
         android:elevation="4dp"
         android:background="@drawable/instruction_text_background"
         app:layout_constraintEnd_toEndOf="parent"
-        app:layout_constraintStart_toEndOf="@+id/linearLayout"
+        app:layout_constraintStart_toEndOf="@+id/instructionLayoutManeuver"
         app:layout_constraintTop_toTopOf="parent">
 
         <TextView
@@ -76,8 +76,8 @@
         layout="@layout/turn_lane_layout"
         android:layout_width="wrap_content"
         android:layout_height="wrap_content"
-        app:layout_constraintStart_toEndOf="@+id/linearLayout"
-        app:layout_constraintTop_toBottomOf="@+id/frameLayout"/>
+        app:layout_constraintStart_toEndOf="@+id/instructionLayoutManeuver"
+        app:layout_constraintTop_toBottomOf="@+id/instructionLayoutText"/>
 
     <LinearLayout
         android:id="@+id/rerouteLayout"

